### PR TITLE
Remove obvious misinformation in 2.x contribution docs.

### PR DESCRIPTION
**Pull request by markstory**
Remove the obvious falsehoods and outdated workflow from the 2.x docs.

Refs #4629

### CODE REVIEW

File: en/contributing/code.rst [hunk 1]
```diff
@@ -112,16 +115,11 @@ branch, as you cannot edit it once the pull request is created.
...
-  should choose the branch with the next version number. For example if the
-  current stable release is ``2.2.2``, the branch accepting new features will be
-  ``2.3``
-* If your change is a breaks existing functionality, or API's then you'll have
-  to choose then next major release. For example, if the current release is
-  ``2.2.2`` then the next time existing behavior can be broken will be in
-  ``3.0`` so you should target that branch.
-
+  should choose the ``2.next`` branch.
+* If your change is a breaks existing functionality, your patch is unlikely to
```
Comment (antograssiot) [inline en/contributing/code.rst]: "Your change breaks "

File: en/contributing/code.rst [hunk 2]
```diff
@@ -112,16 +115,11 @@ branch, as you cannot edit it once the pull request is created.
...
+  be merged into 2.x. Instead you should target 4.0
```
Comment (antograssiot) [inline en/contributing/code.rst]: Do we have a 4.0 branch yet ?

**Comment by antograssiot** [comment]
Looks good

File: en/contributing/code.rst (hunk 2 above)
Comment (markstory) [inline en/contributing/code.rst]: Nope

File: en/contributing/code.rst (hunk 1 above)
Comment (markstory) [inline en/contributing/code.rst]: Thanks :blush:

**Pull request merged**